windows/README_MINGW

```diff
@@ -0,0 +1,218 @@
+
+There are prebuilt MinGW binaries for all the libraries that can be installed
+using the pacman package manager, but since we are using a compiler that is not
+the default version shipped by MinGW, it seems better idea to build from source
+to avoid any linker and runtime issues. A good technical explanation and/or
+evidence for this statement is currently missing from this guide.
+
+Fortunately there are PKGBUILD files for doing so together with a nice guide at
+https://github.com/msys2/MINGW-packages
+
+git clone https://github.com/msys2/MINGW-packages.git
+
+cd MINGW-packages/mingw-w64-{libname}-git
+MINGW_INSTALLS=mingw64 makepkg-mingw -sLf
+pacman -U mingw-w64-{libname}-git-{suffix}.pkg.tar.xz
+
+Repeat procedure for each library replacing {libname} with the appropriate name
+and {suffix} with whatever the above process created.
+
+db
+libsndfile
+libsamplerate
+
+Some libraries like libsndfile and libsamplerate will ask for installing extra
+dependencies, it is ok to do so. The following need to be built in order:
+
+libtre
+libsystre
+
+libsystre is a wrapper around libtre that allows including <regex.h> later.
+Before building it adjust the includes path, it can be unset afterwards:
+
+export C_INCLUDE_PATH=/mingw64/include
+
+Finally for portaudio the Steinberg ASIO SDK should be manually downloaded.
+It can be found at https://www.steinberg.net/en/company/developers.html. The waf
+script will later check if the SDK is present at /opt/asiosdk
+
+wget https://www.steinberg.net/asiosdk -O /tmp/asiosdk.zip
+unzip /tmp/asiosdk.zip -d /tmp
+mv /tmp/asiosdk_{version}_{date} /opt/asiosdk
+
+The PKGBUILD description file needs to be patched so configure is called with
+the necessary flags for ASIO:
+
+--with-asiodir=/opt/asiosdk   <- new option     
+--with-winapi=wmme,directx,wasapi,vdmks,asio   <- append 'asio' to existing
+
+Currently the shared library version of portaudio fails to build, a further
+patch is needed so only the static version is generated. Comment out lines
+[61-75], [82-83] from PKGBUILD.
+
+
+
+Compiling JACK
+--------------
+
+- Clone repo
+
+git clone https://github.com/jackaudio/jack2
+
+- Build and install
+
+cd jack2
+./waf configure --prefix=/opt/jack && ./waf -p install
+
+The resulting files can be found at /c/opt/jack/bin, or c:\msys64\opt\jack\bin
+
+
+
+Compiling JackRouter
+--------------------
+
+Assuming you have Visual Studio installed, just open and build the project:
+
+windows/JackRouter/JackRouter_MinGW_deps.vcxproj
+
+Differences from the original JackRouter.vcxproj:
+
+- Depends on the asiosdk files from the previous MinGW JACK installation (i.e.,
+  it points to absolute paths starting with c:\msys64\opt)
+- Links to the jack.dll.a created by the previous build, instead of libjack.lib
+  This means JackRouter.dll will currently depend on libjack-0.dll
+- Force includes stdint.h and defines _STDIN_H to avoid int8_t, int32_t and
+  uint32_t basic types redefinition duringn compilation
```

Comment:
Using VS2017 15.9.16 and the Windows SDK 10.0.17134.0, I'm unable to build the project found in `windows\JackRouter\JackRouter_MinGW_deps.vcxproj` (from your forked repository) due to redefinition of int8_t, int32_t and uint32_t. Configuration is `Debug` and `x64`. Interestingly, I'm unable to find the mentioned `_STDIN_H` in the entire project solution with the VS search function.

Reply:
Sorry, there is an unfortunate typo in the guide, it is `_STDINT_H` and not `_STDIN_H`:

```
$ grep -R "_STDINT_H" .
./JackRouter_MinGW_deps.vcxproj:      <PreprocessorDefinitions>_STDINT_H;WIN32;NDEBUG;_WINDOWS;PSAPI_VERSION=2;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

By the time I wrote the guide I used an existing VS installation. I still have not tried to build JackRouter totally from scratch (including installing VS), and this `_STDINT_H` thing is really a quick hack rather than a final solution. The latter would involve building JackRouter using a waf script, but that might not be trivial because it involves transitioning from MSVC to GCC.

Let me retry as soon as I get to a Windows machine again. Meanwhile you can try commenting out any redefinition of `int8_t`, `int32_t` and `uint32_t`. That is ugly as sin but should work.

Reply:
The preprocessor definiton was missing in the `Debug` configuration (which is the default VS setting when opening a project for the first time). The configuration `Release` works. I fixed these settings with this commit https://github.com/Schroedingers-Cat/jack2/commit/91d54ecca3fa603d2742de53bc96fd77eff6c282 but I'm unable to PR into your fork. You might also want to consider including this commit https://github.com/Schroedingers-Cat/jack2/commit/91d54ecca3fa603d2742de53bc96fd77eff6c282 which ignores VS generated user files and build files from version control.  

Additionally, you'll need to install the two marked entries from the VS installer in order to succesfully build: 
![grafik](https://user-images.githubusercontent.com/6382400/64939732-819cd380-d851-11e9-801f-6de147c30770.png)

The left circled selection is rather obvious but the one to the right is an additional selection you have to make manually.

Reply:
Awesome finding － yes, I forgot I only worked on the Release variant. Thanks for the update, maybe you cannot initiate the PR because you forked the original Jack2 repo and not my own fork? . It is correct MFC needs to be installed in order for the build to succeed, I am not sure why (I am not a proficient Windows programmer) but think I read that somewhere and did not include the step in the guide, so will update accordingly. Thanks.

Reply:
Turns out that my fork was a bit outdated and that's propably why github hid your repo when creating a PR. That's now fixed and you can just accept the PR and have the changes included here automatically.

Reply:
Merged, thanks.